docs/extending/localtunneldebugger.md

```diff
@@ -0,0 +1,143 @@
+Providers must be hosted within a Visual Studio Code extension.  This table summarises what
+your providers and their hosting extension need to do; the rest of this article goes into detail.
+
+| Component            | Responsibilities                                                     |
+|----------------------|----------------------------------------------------------------------|
+| Your extension       | <ul><li>Activate when the Cluster Explorer is displayed or when the Debug (Local Tunnel) option is selected in the command palette</li><li>Register debug providers with Kubernetes extension</li><li>Tag itself as a local tunnel debug provider in the Visual Studio Marketplace</li></ul>        |
+| Local Tunnel Debug Provider| <ul><li>Implement the local tunnel debug provider interface</li><li>Resolve the command target</li><li>Gather choices from the user in order to generate or select the debug configuration to use</li><li>Start a Local Tunnel debugging session in response to the Debug (Local Tunnel) command</li></ul>  |
+| Kubernetes extension | <ul><li>Display the Debug (Local Tunnel) command in the palette or when interacting with supported resources in the Cluster Explorer</li><li>Allow users to configure which debug provider to use</li></ul>                               |                              |
+
+## Implementing the local tunnel debug provider
+
+A local tunnel debug provider must implement the following interface.  (For documentation purposes
+the interface is written in TypeScript terms but any JavaScript object that provides
+the specified properties and methods will do.)
+
+```javascript
+interface LocalTunnelDebugProvider {
+    readonly id: string;
+    startDebugging(target?: any): void;
+}
+```
+
+### Implementing the metadata
+
+The `id` should be a user-friendly name for your debug provider, such as `Contoso Trampoline Debugger`.  
+The `id` will allow users to configure a default local tunnel debugger to use by setting the `vs-kubernetes.local-tunnel-debug-provider` field under `vs-kubernetes` in settings.json.
+
+### Implementing the debugger
+
+Your provider's implementation of `startDebugging(target?: any)` is responsible for resolving the debug target (such as a pod or service) and starting the debug session. You can use the cluster explorer API to resolve the target (see [commandtargets.md](commandtargets.md) for more info). An example implementation is given below:
+
+```javascript
+const MY_PROVIDER = {
+    id: "Contoso Trampoline Debugger",
+    startDebugging: (target?: any) => startDebugSession(target)
+};
+
+function startDebugSession(target?: any) {
+    const clusterExplorer = await k8s.extension.clusterExplorer.v1;
+    const commandTarget = clusterExplorer.resolveCommandTarget(target);
+
+    if (commandTarget) {
+        if (commandTarget.nodeType === 'resource') {
+            if (commandTarget.resourceKind.manifestKind === 'Pod') {
```

Comment:
It would be good to mention in this section what kinds of resources a provider implementation should handle.  But I wonder if we need to put that burden on providers - e.g. will every provider need the same logic and UI for going from a Deployment to a Pod, and if so can we provide this in the core?

Reply:
Ooh, good point. I will definitely add which resources the provider implementation is expected to handle. Regarding whether or not to provide this in the core, let me sync offline about @daniv-msft about that, but if in practice every provider would be reimplementing the same logic to resolve the pod behind a deployment then that sounds reasonable to provide it up front.

Reply:
I synced with Dan offline, and we felt that at the moment it might not make sense to offer this in the core of the extension, as different providers could have different preferences regarding how to resolve a pod from a deployment. However this is definitely up for discussion, if you or the team have different thoughts about it!